src/NaNMath.jl

```diff
@@ -61,6 +61,48 @@ function sum(x::AbstractArray{T}) where T<:AbstractFloat
     return result
 end
 
+"""
+NaNMath.median(A)
+
+##### Args:
+* `A`: An array of floating point numbers
+
+##### Returns:
+*   Returns the median of all elements in the array, ignoring NaN's.
+    Returns NaN for an empty array or array containing NaNs only.
+
+##### Examples:
+```julia
+using NaNMath
+NaNMath.median([1., 2., 3., NaN]) # result: 2.
+NaNMath.median([1., 2., NaN]) # result: 1.5
+NaNMath.median([NaN]) # result: NaN
+```
+"""
+function median(x::AbstractArray{T}) where T<:AbstractFloat
+
+    if ndims(x) > 1
```

Comment:
rather than check `ndims` we can dispatch on the type of the `AbstractArray`, since the dimensions are included in the type, i.e.
```julia
median(x::AbstractArray{<:AbstractFloat}) = median(collect(Iterators.flatten(x)))

function median(x::AbstractVector{<:AbstractFloat})
   x = sort(filter(!isnan, x)
   ...
   ...
end
```
(`AbstractVector` is an alias for `AbstractArray{T, 1}`, and we let the non-Vector case hit the generic `AbstractArray{T}` method which first creates a Vector and then just uses the Vector method)

Reply:
Sweet, that's really elegant!

I guess the `where T<:AbstractFloat` is only used when you need `T` in the body of the function?